### DISCUSSION TITLE
How to get detailed goal information via the Stats API

### DISCUSSION THREAD

**Discussion by waldhacker-joerg** [plausible/analytics#1509] [Self-Hosted Support]
Hi, I'm trying to get the Goal Conversions data via the Stats API, especially the total conversions. I found out that you can get the available goals and the unique conversions via `api/v1/stats/breakdown` and `property=event:goal` or `property=event:name`. But how can I get the data for _Total_ and _CR_ using the Stats API?

![Screenshot from 2021-12-06 00-59-12](https://user-images.githubusercontent.com/92523446/144769524-471bf9de-5b45-46f0-9dea-17d9e0bbe41c.png)

**Comment by nimahajan** [comment]
I am also stuck here.  I can get uniques but not able to get total

**Reply by mbucak** [reply]
did you solve this issue

**Comment by ukutaht** [comment]
To get the total events, there is an undocumented `events` metric that you should be able to query.

For the conversion rate you have to make a separate query to get the total visitors (from the aggregate endpoint, without the goal filter) and then do the calculation on your side. That's what the frontend does.

**Reply by waldhacker-joerg** [reply]
Thank you for the answer. I had seen the metric in the code, but unfortunately it does not work for me.
If I execute the following command:
`curl -H "Authorization: Bearer ${TOKEN}" "https://plausible.site/api/v1/stats/breakdown?site_id=$SITE_ID&property=event:name&metrics=visitors,events"`
I get the error message:
`{"error":"The metric 'events' is not recognized. Find valid metrics from the documentation: https://plausible.io/docs/stats-api#get-apiv1statsbreakdown"}`

The same happens with `property=event:goal`

I use the docker image plausible/analytics:latest

**Reply by ukutaht** [reply]
Ah OK. This means the `events` metric is available internally but just hasn't been added to the public API yet. The API controller should accept the `events` metric and just pass it on to the DB layer.

I'm sorry I'm busy with other things at the moment but happy to accept a PR for this!

**Reply by waldhacker1** [reply]
I have created a PR for this: https://github.com/plausible/analytics/pull/1529

**Reply by ukutaht** [reply]
Thanks @waldhacker1! This is now merged, will be available on https://plausible.io tomorrow and in the next self-hosted release.

**Comment by metmarkosaric** [comment]
**Update:** `conversion_rate` is now available in the stats API too. See https://plausible.io/docs/stats-api#metrics
